### PR TITLE
patchShebangs: Allow multiple args

### PR DESCRIPTION
**Pull request by JohnAZoidberg**
###### Motivation for this change
It's tempting to think patchShebangs supports multiple arguments.
Without this patch it just silently ignores all but the first. Now it
patches the shebangs in all of its arguments.

Fixes: #57695

Obviously I haven't tested it on nixpkgs because it rebuilds *everything*. Please check whether there might be some edge cases that I forgot which could break with these changes.
I tried the function on its own (copied tothe `patchPhase` of a derivation) and it seems to work as intended.

Seems like there's no docbook documentation for this function. I think I'll do that next.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/build-support/setup-hooks/patch-shebangs.sh
Comment (matthewbauer) [inline pkgs/build-support/setup-hooks/patch-shebangs.sh]: Could you just put ```find "$@"``` here? I think that could be more efficient

Comment (JohnAZoidberg) [reply pkgs/build-support/setup-hooks/patch-shebangs.sh]: Oh, that would remove the need for my loop, right?

Comment (matthewbauer) [reply pkgs/build-support/setup-hooks/patch-shebangs.sh]: Yeah would just have to remove the ```[ -e "$dir" ] || return 0``` part as well.

Comment (JohnAZoidberg) [reply pkgs/build-support/setup-hooks/patch-shebangs.sh]: Thanks! I tried the function out on its own and it seems to be doing the right things.

**Comment by FRidh** [comment]
Waiting for reviews before merging.

**Review by matthewbauer — APPROVED** [review]

File: pkgs/build-support/setup-hooks/patch-shebangs.sh
Comment (chreekat) [inline pkgs/build-support/setup-hooks/patch-shebangs.sh]: This change, combined with the removal of line 43, means the function will fail clumsily if no arguments are passed to it. But considering that the old version silently and surprisingly ignores nonexistent files, this could be considered an improvement. ;) However, since it's a change in behavior, might it cause build failures? Should we let them happen? (Serious question.)

Comment (JohnAZoidberg) [reply pkgs/build-support/setup-hooks/patch-shebangs.sh]: What about we check if there are no arguments supplied in line 43?
```bash
  if [ $# -eq 0 ]; then
      echo "No arguments supplied to patchShebangs" >&2
      return 3141592  # Not sure which return code is best, maybe just 1
  fi
```

**Comment by JohnAZoidberg** [comment]
@matthewbauer Do you still approve?
Does anyone else have any thoughts?
@chreekat Does this address your concerns?

**Comment by chreekat** [comment]
Lgtm, yep

File: pkgs/build-support/setup-hooks/patch-shebangs.sh
Comment (JohnAZoidberg) [reply pkgs/build-support/setup-hooks/patch-shebangs.sh]: I made it exit with 0 like it was before so that existing scripts don't break.

File: pkgs/build-support/setup-hooks/patch-shebangs.sh
Comment (matthewbauer) [inline pkgs/build-support/setup-hooks/patch-shebangs.sh]: This should be >0 so bash recognizes it as an error

Comment (matthewbauer) [reply pkgs/build-support/setup-hooks/patch-shebangs.sh]: Should be return actually

File: pkgs/build-support/setup-hooks/patch-shebangs.sh
Comment (matthewbauer) [inline pkgs/build-support/setup-hooks/patch-shebangs.sh]: I think header should be kept too

Comment (JohnAZoidberg) [reply pkgs/build-support/setup-hooks/patch-shebangs.sh]: But it just calls echo and is marked as obsolete. https://github.com/NixOS/nixpkgs/blob/master/pkgs/stdenv/generic/setup.sh#L115

File: pkgs/build-support/setup-hooks/patch-shebangs.sh
Comment (Infinisil) [reply pkgs/build-support/setup-hooks/patch-shebangs.sh]: Why `>0`? That just pipes the string to a file with name "0"?

**Comment by Infinisil** [comment]
https://github.com/NixOS/nixpkgs/pull/60559#discussion_r287837965 ?

File: pkgs/build-support/setup-hooks/patch-shebangs.sh
Comment (matthewbauer) [reply pkgs/build-support/setup-hooks/patch-shebangs.sh]: Oops meant "greater than 0". That is non-zero code.

Comment (JohnAZoidberg) [reply pkgs/build-support/setup-hooks/patch-shebangs.sh]: ohhhh

**Pull request merged**
